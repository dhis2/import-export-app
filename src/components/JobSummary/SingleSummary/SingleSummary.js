import i18n from '@dhis2/d2-i18n'
import {
    Table,
    TableBody,
    TableCell,
    TableHead,
    TableRow,
    TableRowHead,
    TableCellHead,
} from '@dhis2/ui'
import PropTypes from 'prop-types'
import React from 'react'
import { FormField } from '../../index.js'
import { statsPropType } from '../helper.js'
import { SingleStatusTable } from '../SingleStatusTable/SingleStatusTable.js'

const SingleSummary = ({
    importType,
    importCount,
    status,
    description,
    conflicts,
    validationReport,
    id,
}) => (
    <div>
        <FormField
            label={id ? `${i18n.t('Summary')} #${id}` : i18n.t('Summary')}
            dataTest="job-summary-single-summary"
            name="summary"
        >
            <>
                {status && description && (
                    <SingleStatusTable
                        description={description}
                        status={status}
                    />
                )}
                <Table>
                    <TableHead>
                        <TableRowHead>
                            <TableCellHead>{i18n.t('Created')}</TableCellHead>
                            <TableCellHead>{i18n.t('Deleted')}</TableCellHead>
                            <TableCellHead>{i18n.t('Ignored')}</TableCellHead>
                            <TableCellHead>{i18n.t('Updated')}</TableCellHead>
                            <TableCellHead>{i18n.t('Total')}</TableCellHead>
                        </TableRowHead>
                    </TableHead>
                    <TableBody>
                        <TableRow>
                            <TableCell>
                                {importCount?.imported ??
                                    importCount?.created ??
                                    '0'}
                            </TableCell>
                            <TableCell>{importCount?.deleted}</TableCell>
                            <TableCell>{importCount?.ignored}</TableCell>
                            <TableCell>{importCount?.updated}</TableCell>
                            <TableCell>{importCount?.total}</TableCell>
                        </TableRow>
                    </TableBody>
                </Table>
            </>
        </FormField>
        {!!validationReport?.errorReports?.length && (
            <FormField
                label={`${i18n.t('Reports')}`}
                dataTest="tracker-summary-reports"
                name="tracker-reports"
            >
                <Table>
                    <TableHead>
                        <TableRowHead>
                            <TableCellHead>{i18n.t('UID')}</TableCellHead>
                            <TableCellHead>
                                {i18n.t('Error Code')}
                            </TableCellHead>
                            <TableCellHead>{i18n.t('Message')}</TableCellHead>
                            <TableCellHead>
                                {i18n.t('Tracker Type')}
                            </TableCellHead>
                            {/* <TableCellHead>{i18n.t('')}</TableCellHead> */}
                        </TableRowHead>
                    </TableHead>
                    <TableBody>
                        {validationReport.errorReports.map((c, i) => (
                            <TableRow
                                key={`job-summary-report-${c.object}-${i}`}
                            >
                                <TableCell>{c.uid}</TableCell>
                                <TableCell>
                                    <a
                                        target="_blank"
                                        rel="noreferrer"
                                        href="https://docs.dhis2.org/en/develop/using-the-api/dhis-core-version-master/tracker.html#webapi_nti_error_codes"
                                    >
                                        {c.warningCode ?? c.errorCode}
                                    </a>
                                </TableCell>
                                <TableCell>{c.message}</TableCell>
                                <TableCell>{c.trackerType}</TableCell>
                            </TableRow>
                        ))}
                    </TableBody>
                </Table>
            </FormField>
        )}
        {conflicts && (
            <FormField
                label={`${i18n.t('Conflicts')}`}
                dataTest="job-summary-conflicts"
                name="conflicts"
            >
                <Table>
                    <TableHead>
                        <TableRowHead>
                            <TableCellHead>
                                {importType !== 'GEOJSON_IMPORT'
                                    ? i18n.t('Object')
                                    : i18n.t('Indexes')}
                            </TableCellHead>
                            <TableCellHead>{i18n.t('Value')}</TableCellHead>
                        </TableRowHead>
                    </TableHead>
                    <TableBody>
                        {conflicts.map((c, i) => (
                            <TableRow
                                key={`job-summary-conflicts-${c.object}-${i}`}
                            >
                                <TableCell>
                                    {importType !== 'GEOJSON_IMPORT'
                                        ? c.object
                                        : (c.indexes || []).join(',')}
                                </TableCell>
                                <TableCell>{c.value}</TableCell>
                            </TableRow>
                        ))}
                    </TableBody>
                </Table>
            </FormField>
        )}
    </div>
)

SingleSummary.propTypes = {
    importCount: statsPropType.isRequired,
    conflicts: PropTypes.arrayOf(
        PropTypes.shape({
            value: PropTypes.string.isRequired,
            indexes: PropTypes.arrayOf(PropTypes.number),
            object: PropTypes.string,
        })
    ),
    description: PropTypes.string,
    id: PropTypes.string,
    importType: PropTypes.string,
    status: PropTypes.string,
    validationReport: PropTypes.shape({
        errorReports: PropTypes.arrayOf(
            PropTypes.shape({
                errorCode: PropTypes.string,
                message: PropTypes.string,
                trackerType: PropTypes.string,
                uid: PropTypes.string,
            })
        ),
        warningReports: PropTypes.arrayOf(
            PropTypes.shape({
                errorCode: PropTypes.string,
                message: PropTypes.string,
                trackerType: PropTypes.string,
                uid: PropTypes.string,
            })
        ),
    }),
}

export { SingleSummary }
