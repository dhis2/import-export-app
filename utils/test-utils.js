import * as testingLibrary from '@testing-library/react'
import * as customQueries from './custom-queries.js'

const { queries } = testingLibrary

console.log('testingLibrary', testingLibrary)

const customRender = (ui, options) =>
    testingLibrary.render(ui, {
        queries: { ...queries, ...customQueries },
        ...options,
    })

/* eslint-disable */
// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
/* eslint-enable */
