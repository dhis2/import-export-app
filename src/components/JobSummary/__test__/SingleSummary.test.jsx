import React from 'react'
import { render } from 'test-utils'
import '@testing-library/jest-dom/extend-expect.js'

import { SingleSummary } from '../SingleSummary/SingleSummary.jsx'

const props = {
    importCount: { imported: 1, deleted: 2, ignored: 3, updated: 4, total: 10 },
    status: 'WARNING',
    description: 'Import process completed successfully',
    conflicts: [
        { object: 'a', value: 'conflict a value' },
        { object: 'b', value: 'conflict b value' },
    ],
    id: '1',
}

it(`matches snapshot`, () => {
    const { asFragment } = render(<SingleSummary {...props} />)
    expect(asFragment()).toMatchSnapshot()
})
