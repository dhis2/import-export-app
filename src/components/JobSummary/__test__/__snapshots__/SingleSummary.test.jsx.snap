// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="container"
      data-test="job-summary-single-summary"
    >
      <span
        class="label"
      >
        Summary #1
      </span>
      <table
        class="jsx-2430604489 "
        data-test="dhis2-uicore-table"
      >
        <thead
          data-test="dhis2-uicore-tablehead"
        >
          <tr
            class="jsx-3124144591 zebraStriping"
            data-test="dhis2-uicore-tablerowhead"
          >
            <th
              class="jsx-1796049005 "
              data-test="dhis2-uicore-tablecellhead"
            >
              Status
            </th>
            <th
              class="jsx-1796049005 "
              data-test="dhis2-uicore-tablecellhead"
            >
              Description
            </th>
          </tr>
        </thead>
        <tbody
          data-test="dhis2-uicore-tablebody"
        >
          <tr
            class="jsx-3124144591 zebraStriping"
            data-test="dhis2-uicore-tablerow"
          >
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              WARNING
            </td>
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              Import process completed successfully
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="jsx-2430604489 "
        data-test="dhis2-uicore-table"
      >
        <thead
          data-test="dhis2-uicore-tablehead"
        >
          <tr
            class="jsx-3124144591 zebraStriping"
            data-test="dhis2-uicore-tablerowhead"
          >
            <th
              class="jsx-1796049005 "
              data-test="dhis2-uicore-tablecellhead"
            >
              Created
            </th>
            <th
              class="jsx-1796049005 "
              data-test="dhis2-uicore-tablecellhead"
            >
              Deleted
            </th>
            <th
              class="jsx-1796049005 "
              data-test="dhis2-uicore-tablecellhead"
            >
              Ignored
            </th>
            <th
              class="jsx-1796049005 "
              data-test="dhis2-uicore-tablecellhead"
            >
              Updated
            </th>
            <th
              class="jsx-1796049005 "
              data-test="dhis2-uicore-tablecellhead"
            >
              Total
            </th>
          </tr>
        </thead>
        <tbody
          data-test="dhis2-uicore-tablebody"
        >
          <tr
            class="jsx-3124144591 zebraStriping"
            data-test="dhis2-uicore-tablerow"
          >
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              1
            </td>
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              2
            </td>
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              3
            </td>
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              4
            </td>
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              10
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="container"
      data-test="job-summary-conflicts"
    >
      <span
        class="label"
      >
        Conflicts
      </span>
      <table
        class="jsx-2430604489 "
        data-test="dhis2-uicore-table"
      >
        <thead
          data-test="dhis2-uicore-tablehead"
        >
          <tr
            class="jsx-3124144591 zebraStriping"
            data-test="dhis2-uicore-tablerowhead"
          >
            <th
              class="jsx-1796049005 "
              data-test="dhis2-uicore-tablecellhead"
            >
              Object
            </th>
            <th
              class="jsx-1796049005 "
              data-test="dhis2-uicore-tablecellhead"
            >
              Value
            </th>
          </tr>
        </thead>
        <tbody
          data-test="dhis2-uicore-tablebody"
        >
          <tr
            class="jsx-3124144591 zebraStriping"
            data-test="dhis2-uicore-tablerow"
          >
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              a
            </td>
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              conflict a value
            </td>
          </tr>
          <tr
            class="jsx-3124144591 zebraStriping"
            data-test="dhis2-uicore-tablerow"
          >
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              b
            </td>
            <td
              class="jsx-996330601 "
              data-test="dhis2-uicore-tablecell"
            >
              conflict b value
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
